src/appleseed/renderer/modeling/phasefunction/henyeyphasefunction.cpp

```diff
@@ -219,11 +219,12 @@ class HenyeyPhaseFunction
     virtual void evaluate_transmission(
         const ShadingRay&     volume_ray,
         const void*           data,
-        float                 distance,
+        const float           distance,
         Spectrum&             spectrum) const APPLESEED_OVERRIDE
     {
         extinction_coefficient(volume_ray, data, distance, spectrum);
-        spectrum = exp(-distance * spectrum);
+        for (int i = 0; i < spectrum.size(); ++i)
```

Comment:
Another thing: it's best to evaluate `spectrum.size()` only once instead of once per iteration: that gives us peace of mind regardless of the optimization decision of the compiler:

    for (size_t i = 0, e = spectrum.size(); i < e; ++i)